### ISSUE TITLE
Under the hood of kubebuilder

### ISSUE THREAD

**Issue by devdattakulkarni** [kubernetes-sigs/kubebuilder#169]
I have written a post describing some of the under the hood details of kubebuilder.

https://medium.com/@cloudark/under-the-hood-of-kubebuilder-framework-ff6b38c10796

@pwittrock Let me know if this is correctly capturing kubebuilder's architecture and its interaction points with custom controllers.

**Comment by droot** [comment]
Thanks @devdattakulkarni . Just reviewed the post. Thanks for the write up, it correctly captures the architecture and interaction between various components.

**Comment by droot** [comment]
@devdattakulkarni FYI, we released 1.0.0 last week (https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v1.0.0) which incorporates feedback from the previous versions. It addresses some of the concerns in your original post about new concept a user has to learn to use Kubebuilder. We have tried to reduce the abstractions a user has to work with or rather made those abstractions easier to grok by providing simple libraries under [controller-runtime](https://godoc.org/sigs.k8s.io/controller-runtime/pkg). 
Give it a [try](http://book.kubebuilder.io).
